src/demos/findex/findex.ts

```diff
@@ -0,0 +1,172 @@
+import { Findex } from "../../interface/findex/findex";
+import { DB } from "./demo_db";
+import { masterKeysFindex } from "./demo_keys";
+import { logger } from "../../utils/logger";
+import { hexDecode, hexEncode, sanitizeString } from "../../utils/utils";
+
+export class FindexDemo {
+  private _db: DB;
+  private _policy: Policy;
+  private _masterKeysCoverCrypt: AbeMasterKey;
+
+  constructor(db: DB, policyAxes: PolicyAxis[], maxAttributeCreations: number) {
+    this._db = db;
+    this._policy = new Policy(policyAxes, maxAttributeCreations)
+    this._masterKeysCoverCrypt = generateMasterKeys(this._policy);
+  }
+
+  public get db(): DB {
+    return this._db;
+  }
+  public get policy(): Policy {
+    return this._policy;
+  }
+  public get masterKeysCoverCrypt(): AbeMasterKey {
+    return this._masterKeysCoverCrypt;
+  }
+
+  async encryptUsers(metadataUid: string) {
+    const policyBytes = this._policy.toJsonEncoded();
+
+    const elements = await this._db.getUsers();
+    for (const element of elements) {
+      const encryptedBasic = coverCryptEncrypt(
+        policyBytes,
+        this._masterKeysCoverCrypt.publicKey,
+        metadataUid,
+        [`department::marketing`, `country::${element.country}`],
+        JSON.stringify({
+          firstName: element.firstName,
+          lastName: element.lastName,
+          country: element.country,
+          region: element.region
+        }))
+
+      const encryptedHr = coverCryptEncrypt(
+        policyBytes,
+        this._masterKeysCoverCrypt.publicKey,
+        metadataUid,
+        [`department::HR`, `country::${element.country}`], JSON.stringify({
+          email: element.email,
+          phone: element.phone,
+          employeeNumber: element.employeeNumber
+        })
+      )
+
+      const encryptedSecurity = coverCryptEncrypt(
+        policyBytes,
+        this._masterKeysCoverCrypt.publicKey,
+        metadataUid,
+        [`department::security`, `country::${element.country}`], JSON.stringify({
+          security: element.security
+        }))
+
+      const upsertedEncElement = await this._db.upsertEncryptedUser({
+        enc_basic: hexEncode(encryptedBasic),
+        enc_hr: hexEncode(encryptedHr),
+        enc_security: hexEncode(encryptedSecurity)
+      });
+
+      await this._db.upsertUserEncUidById(element.id, { enc_uid: upsertedEncElement[0].uid });
+    };
+
+  }
+
+  async resetAndUpsert(location: string) {
+    await this._db.deleteAllChainTableEntries();
+    await this._db.deleteAllEntryTableEntries();
+    type Element = { [key: string]: string; };
+    const elements: Element[] = await this._db.getUsers();
+    const sanitizedElements: Element[] = elements.map((element) => {
+      Object.keys(element).forEach((key) => {
+        if (element[key]) {
+          element[key] = sanitizeString(element[key])
+        }
+      });
+      return element;
+    })
```

Comment:
You can remove the `type Element` by using `keyof typeof element` in TypeScript. This avoid creating a new type.
```suggestion
    const sanitizedElements = elements.map((element) => {
      let key: keyof typeof element;
      for (key in element) {
        element[key] = sanitizeString(element[key])
      }
      return element;
    })
```

Reply:
partially done. User type is used later on